src/main/java/org/minions/devfund/katerinaanzoleaga/battleship/Ship.java

```diff
@@ -0,0 +1,278 @@
+     * @param column int
+     * @param horizontal bool
+     * @param ocean ocean
+     * @return True if it is possible to place a ship in the defined position.
+     */
+    public boolean okToPlaceShipAt(int row, int column, boolean horizontal, final Ocean ocean) {
+        if (ocean.isValidPosition(row, column)) {
+            if (horizontal) {
+                if (fitsInDirection(column, ocean)) {
+                    for (int i = row - 1; i <= row + 1; ++i) {
+                        for (int j = column - 1; j <= column + this.getLength(); ++j) {
+                            if (ocean.isValidPosition(i, j)) {
+                                if (ocean.isOccupied(i, j)) {
+                                    return false;
+                                }
+                            }
+                        }
+                    }
+                    return true;
+                } else {
+                    return false;
+                }
+
+            } else {
+                if (fitsInDirection(row, ocean)) {
+                    for (int i = row - 1; i <= row + this.getLength(); ++i) {
+                        for (int j = column - 1; j <= column + 1; ++j) {
+                            if (ocean.isValidPosition(i, j)) {
+                                if (ocean.isOccupied(i, j)) {
+                                    return false;
+                                }
+                            }
+                        }
+                    }
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+        } else {
+            return false;
+        }
+    }
+
+
+    /**
+     * Places a ship at the position specified and with the direction specified.
+     * @param row int
+     * @param column int
+     * @param horizontal bool
+     * @param ocean ocean
+     */
+    public void placeShipAt(int row, int column, boolean horizontal, final Ocean ocean) {
+        if (okToPlaceShipAt(row, column, horizontal, ocean)) {
+            this.bowRow = row;
+            this.bowColumn = column;
+            this.horizontal = horizontal;
+            Ship[][] newShip = ocean.getShipArray();
+            if (horizontal) {
+                for (int i = column; i < column + this.length; ++i) {
+                    newShip[row][i] = this;
+                }
+                ocean.setShips(newShip);
+            } else {
+                for (int i = row; i < row + this.length; ++i) {
+                    newShip[i][column] = this;
+                }
+            }
+            ocean.setShips(newShip);
+        }
+    }
+
+    /**
+     * This returns -1 if the position does not belong to the ship.
+     * Returns the relative position of the hit i shot at that position.
+     * @param row int
+     * @param column int
+     * @return Returns the relative position of the hit i shot at that position.
+     */
+    public int isInShipPosition(int row, int column) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)